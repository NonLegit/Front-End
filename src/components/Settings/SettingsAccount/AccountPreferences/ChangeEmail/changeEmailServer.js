// import { removeRedditCookie } from '../../../Authentication/authenticationServer';
import { redirectHome } from '../../../../../utils/Redirect';
import axios from '../../../../../services/instance';
// import { redirectLogin } from '../../../../../utils/Redirect';
/**
 * - pchange Email of user
 */
export const changeEmail = (newEmail, password) => {
  console.log('change EMAIL', newEmail, password);
  axios.post(
    '/users/change_email',
    {
      newEmail,
      password,
    },
  ).then((response) => {
    console.log('users/change_email', response);
    if (response?.status === 204) {
      alert('Email Updated Sucessfully You will recieve A verification Email');
      redirectHome(100);
    }
  }).catch((error) => {
    // 404
    console.log(error);
    alert(error?.response?.data?.errorMessage);
  });
};
