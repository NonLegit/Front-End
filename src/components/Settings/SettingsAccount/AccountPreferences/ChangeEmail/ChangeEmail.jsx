import CloseIcon from '@mui/icons-material/Close';
import EmailIcon from '@mui/icons-material/Email';
import PriorityHighIcon from '@mui/icons-material/PriorityHigh';
import { Box } from '@mui/system';
import { useEffect, useState } from 'react';
import { EmailFormat } from '../../../../../utils/checkEmail';
import theme from '../../../../../styles/theme';
import {
  IconClose, Header, HeaderIcon, IconError, Container,
  Pragraph, Input, Footer, SaveBtn, Error,
} from './styles';
import { changeEmail } from './changeEmailServer';
/**
 * -Change Email
 *  @component
 *   @property {Boolean} checkEamil - bool to check email is true or not
 *  @property {Boolean} condition -bool to check if input is valid or not
 *  @property {Boolean} error  - bool if there are error or not
 *  @property {String} email  - input email from user
 *  @property {String} pass  - input passworf from user
 *  @return {React.Component} -Change Email popup
 */
function ChangeEmail({ setOpenEmail }) {
  const [condition, setCondition] = useState(false);
  const [checkEamil, setCheckEamil] = useState(false);
  const [error, setError] = useState(false);
  const [pass, setPass] = useState('');
  const [email, setEamil] = useState('');
  useEffect(() => {
    if (pass.length > 0 && checkEamil) {
      setCondition(true);
      console.log('saa');
    } else {
      setCondition(false);
    }
  }, [pass, email, checkEamil]);
  const checkSubmission = () => {
    // Call BackEd Point
    console.log('Function called');
    changeEmail(email, pass);
  };
  return (
    <Container>
      <Header>
        <HeaderIcon>
          <EmailIcon fontSize="large" color="primary" />
          <IconError>
            <PriorityHighIcon fontSize="small" color={theme.palette.neutral.main} />
          </IconError>
        </HeaderIcon>
        Update your email
        <IconClose
          aria-label="close"
          onClick={() => {
            setOpenEmail(false);
            setPass('');
            setEamil('');
            setCheckEamil(false);
            setError(false);
            setCondition(false);
          }}
        >
          <CloseIcon />
        </IconClose>

      </Header>
      <Pragraph>Update your email below. There will be a new verification email sent that you will need to use to verify this new email.</Pragraph>
      <Box>
        <Input
          value={pass}
          onChange={(e) => {
            setPass(e.currentTarget.value.trim());
          }}
          placeholder="CURRENT PASSWORD"
        />
      </Box>
      <Box>
        <Input
          error={error}
          value={email}
          onChange={(e) => {
            setEamil(e.currentTarget.value.trim());
            if (EmailFormat(e.currentTarget.value.trim())) {
              setCheckEamil(true);
            } else {
              setCheckEamil(false);
            }
          }}
          onBlur={() => {
            setError(!checkEamil);
          }}
          placeholder="NEW EMAIL"
        />
      </Box>
      {
              error && (
              <Error>
                Please enter a valid email
              </Error>
              )
          }
      <Footer>
        <SaveBtn variant="contained" condition={condition} onClick={() => checkSubmission()}>Save email</SaveBtn>
      </Footer>
    </Container>
  );
}

export default ChangeEmail;
