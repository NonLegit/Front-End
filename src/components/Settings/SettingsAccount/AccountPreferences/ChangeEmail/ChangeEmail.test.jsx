import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ChangeEmail from './ChangeEmail';
import SettingsProvider from '../../../../../contexts/SettingsProvider';

const mockAlert = jest.fn();
jest.mock('react-alert', () => ({
  ...jest.requireActual('react-alert'),
  useAlert: () => mockAlert,
}));
// test snapshot
test('test snapshot', async () => {
  const setOpenEmail = (flag) => {
    console.log(flag);
  };
  const props = {
    setOpenEmail,
  };
  const tree = renderer.create(
    <Router>
      <SettingsProvider>
        <ChangeEmail {...props} />
      </SettingsProvider>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
