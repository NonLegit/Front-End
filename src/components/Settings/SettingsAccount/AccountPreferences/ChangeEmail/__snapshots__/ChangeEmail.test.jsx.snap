// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1s0gywi-MuiTypography-root-MuiDialogTitle-root"
>
  <div
    className="css-1t1xcwn"
  >
    <div
      className="css-6xiis1"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
        data-testid="EmailIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
        />
      </svg>
      <div
        className="css-w2t8pj"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-color#978A8C MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="PriorityHighIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="19"
            r="2"
          />
          <path
            d="M10 3h4v12h-4z"
          />
        </svg>
      </div>
    </div>
    Update your email
    <button
      aria-label="close"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1y8k0vr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
  <p
    className="css-1f97iu9"
  >
    Update your email below. There will be a new verification email sent that you will need to use to verify this new email.
  </p>
  <div
    className="MuiBox-root css-0"
  >
    <input
      className="css-ht6vgj"
      onChange={[Function]}
      placeholder="CURRENT PASSWORD"
      value=""
    />
  </div>
  <div
    className="MuiBox-root css-0"
  >
    <input
      className="css-ht6vgj"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="NEW EMAIL"
      value=""
    />
  </div>
  <div
    className="css-1jh72zi"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1j2c57i-MuiButtonBase-root-MuiButton-root"
      condition={false}
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save email
    </button>
  </div>
</h2>
`;
